pkg/strvals/parser.go

```diff
@@ -152,6 +169,73 @@ func set(data map[string]interface{}, key string, val interface{}) {
 	data[key] = val
 }
 
+func setIndex(list []interface{}, index int, val interface{}) []interface{} {
+	if len(list) <= index {
+		newlist := make([]interface{}, index+1)
+		for i, v := range list {
```

Comment:
fyi, could use `copy` rather than a loop
https://blog.golang.org/go-slices-usage-and-internals#TOC_5.